src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -746,5 +751,49 @@ public static ScreenshotEvent onScreenshot(BufferedImage image, File screenshotF
         MinecraftForge.EVENT_BUS.post(event);
         return event;
     }
+    
+    public static List<List<BakedQuad>> sortQuadsByIndex(IBakedModel model)
+    {
+        List<List<BakedQuad>> layers = new ArrayList();
+        List<BakedQuad> quads = new ArrayList();
+        
+        for (EnumFacing dir : EnumFacing.values())
+        {
+            quads.addAll(model.getQuads(null, dir, 0));
+        }
+        quads.addAll(model.getQuads(null, null, 0));
+        
+        for (BakedQuad quad : quads)
+        {
+            int index = quad.hasTintIndex() ? quad.getTintIndex() : 0;
+            if (index >= layers.size())
+                layers.add(index, new ArrayList());
+            
+            layers.get(index).add(quad);
+        }
+        return layers;
```

Comment:
why can't this just be a `List<BakedQuad>` sorted using a comparator that compares tint index?